app/modules/asset/service.server.ts

```diff
@@ -212,6 +212,9 @@ export async function getAssets({
       message: "booking dates are needed to hide unavailable assets",
     });
   }
+  if (bookingFrom && bookingTo && where.asset) {
```

Comment:
@mahendraHegde can we put a comment here what is happening. This function handles lots of cases and it will help a lot to other developers as well.